File: backend/types/setting.go
package types

type Registry struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	URL       string `json:"url"`
	IsDefault string `json:"isDefault"`
	Username  string `json:"username"`
	Password  string `json:"password"`
}

type ConfigType int

var (
	ConfigTypeStatic ConfigType = 1
	ConfigTypeVolume ConfigType = 2
)

type Config struct {
	ConfigId    string     `json:"configId"`
	ConfigName  string     `json:"configName"`
	Description string     `json:"description"`
	ConfigType  ConfigType `json:"configType"`
	ConfigValue string     `json:"configValue"`
	CreatedAt   int64      `json:"createdAt"`
	UpdatedAt   int64      `json:"updatedAt"`
}

type Template struct {
	TemplateID   string      `json:"templateId"`
	TemplateName string      `json:"templateName"`
	Content      interface{} `json:"content"`
	CreateAt     int64       `json:"createAt"`
	UpdateAt     int64       `json:"updateAt"`
}

File: front/projects/web/src/types/setting.ts
import { ConfigType } from "@/models"
import { BaseInfo, NewBaseEmptyInfo } from "#/base.ts"

export interface ConfigInfo extends BaseInfo {
  configId: string
  configName: string
  description: string
  configValue: string
  configType: number
}

export function NewConfigEmptyInfo(): ConfigInfo {
  return {
    ...NewBaseEmptyInfo(),
    configId: "",
    configName: "",
    description: "",
    configValue: "",
    configType: ConfigType.Static
  }
}

export interface RegistryInfo extends BaseInfo {
  registryId: string
  registryName: string
  url: string
  isDefault: boolean
  username: string
  password: string
  hasAuth?: boolean
}

export function NewRegistryEmptyInfo(): RegistryInfo {
  return {
    ...NewBaseEmptyInfo(),
    registryId: "",
    registryName: "",
    url: "",
    isDefault: false,
    username: "",
    password: ""
  }
}

export interface NodeInfo extends BaseInfo {
  nodeId: string
  name: string
  ipAddress: string
  port: number
  status: string
  isEnable: boolean
  cpuUsage: number
  memoryUsage: number
  labels: { [key: string]: string }
}

export function NewNodeEmptyInfo(): NodeInfo {
  return {
    ...NewBaseEmptyInfo(),
    nodeId: "",
    name: "",
    ipAddress: "",
    port: 0,
    status: "",
    isEnable: false,
    cpuUsage: 0,
    memoryUsage: 0,
    labels: {}
  }
}
